CartingService/Messaging/MessagingService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.Json;
+using Confluent.Kafka;
+using Microsoft.Extensions.Logging;
+
+public class MessagingService
+{
+    private readonly IItemService _itemService;
+    private readonly ILogger _logger;
+
+    public MessagingService(IItemService itemService, ILogger<MessagingService> logger)
+    {
+        _itemService = itemService;
+        _logger = logger;
+    }
+
+    public async Task Consume(string server, string groupId, CancellationToken cancellationToken)
+    {
+        using var consumer = new ConsumerBuilder<Ignore, string>(new ConsumerConfig { GroupId = groupId, BootstrapServers = server }).Build();
+
+        consumer.Subscribe(nameof(ItemUpdatedMessage));
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            var message = consumer.Consume(cancellationToken);
+            var messageHandled = false;
+            while (!messageHandled)
+            {
+                try
+                {
+                    if (message.Topic == nameof(ItemUpdatedMessage))
```

Comment:
I know that we don't have any other requirements to handle any other message, but the current implementation is not extendable. You would need to modify this class if you would add a new message type. Can you think about some approach to don't need to touch this class in case of adding new messaged (see MediatR if don't have any ideas)

Reply:
Fixed